src/nvim/ex_cmds.c

```diff
@@ -5836,3 +5873,166 @@ void set_context_in_sign_cmd(expand_T *xp, char_u *arg)
     }
   }
 }
+
+/// live_sub()
+/// Open a window for future displaying of the live_sub mode.
+/// 
+/// Does not allow editing in the window. 
+/// Returns when the window is closed.
+///
+/// @param sub the replacement word
+/// @param lmatch the list containing our data
+///
+/// @returns  CR	      if the command is to be executed
+///           Ctrl_C    if it is to be abandoned
+///           K_IGNORE  if editing continues
+
+int ex_window_live_sub(char_u* sub, klist_t(matchedline_T) *lmatch)
+{
+  int i;
+  garray_T winsizes;
+  char_u typestr[2];
+  int save_restart_edit = restart_edit;
+  int save_State = State;
+  int save_exmode = exmode_active;
+  int save_cmdmsg_rl = cmdmsg_rl;
+
+  // Can't do this recursively.  Can't do it when typing a password. 
+  if (cmdwin_type != 0
+      || cmdline_star > 0
+      ) {
+    beep_flush();
+    return K_IGNORE;
+  }
+
+  // Save current window sizes. 
+  win_size_save(&winsizes);
+
+  // Save the current window to restore it later 
+  win_T* oldwin = curwin;
+
+  // Don't execute autocommands while creating the window. 
+  block_autocmds();
+  // don't use a new tab page 
+  cmdmod.tab = 0;
+
+  // close last buffer used for ex_window_live_sub() 
+  buf_T* oldbuf;
+  if((oldbuf = buflist_findname_exp((char_u *)"[live_sub]"))!=NULL) {
+    close_windows (oldbuf, FALSE);
+    close_buffer (NULL, oldbuf, DOBUF_WIPE, FALSE);
+  }
+
+  // Create a window for the command-line buffer. 
+  if (win_split((int)p_cwh, WSP_BOT) == FAIL) {
+    beep_flush();
+    unblock_autocmds();
+    return K_IGNORE;
+  }
+  cmdwin_type = get_cmdline_type();
+
+  // Create the command-line buffer empty. 
+  (void)do_ecmd(0, NULL, NULL, NULL, ECMD_ONE, ECMD_HIDE, NULL);
+  (void)setfname(curbuf, (char_u *)"[live_sub]", NULL, TRUE);
+  set_option_value((char_u *)"bt", 0L, (char_u *)"nofile", OPT_LOCAL);
+  set_option_value((char_u *)"swf", 0L, NULL, OPT_LOCAL);
+  curbuf->b_p_ma = FALSE; // Not Modifiable
+  curwin->w_p_fen = FALSE;
+  curwin->w_p_rl = cmdmsg_rl;
+  cmdmsg_rl = FALSE;
+  RESET_BINDING(curwin);
+
+  // Do execute autocommands for setting the filetype (load syntax). 
+  unblock_autocmds();
+
+  // Showing the prompt may have set need_wait_return, reset it. 
+  need_wait_return = FALSE;
+
+  // Reset 'textwidth' after setting 'filetype' (the Vim filetype plugin sets 'textwidth' to 78). 
+  curbuf->b_p_tw = 0;
+
+  // Initialize line and highliht variables 
+  int line = 0;
+  int src_id_highlight = 0;
+  long match_size = strlen((char*)sub);
+  
+  // allocate a line sized for the window
+  char *str = xmalloc((size_t )curwin->w_frame->fr_width);
+  
+  // Append the lines to our buffer
+  kl_iter(matchedline_T, lmatch, current) {
+    matchedline_T mat = (*current)->data;
+    size_t line_size = sizeof(mat.line) + sizeof(long) + 5;
+
+    // Reallocation if str not long enough
+    if (line_size > curwin->w_frame->fr_width*sizeof(char))
+      assert(xrealloc(str, line_size) != NULL);
```

Comment:
The assert not *can* be removed, it *must* be removed. In some build types with glibc (if NDEBUG is defined) the whole expression will be replaced with `(void) (0)`, effectively removing `xrealloc()` call.